oxidation/crates/parsec_api_types/src/manifest.rs

```diff
@@ -1,15 +1,71 @@
 // Parsec Cloud (https://parsec.cloud) Copyright (c) BSLv1.1 (eventually AGPLv3) 2016-2021 Scille SAS
 
-use chrono::prelude::*;
-use serde::{Deserialize, Serialize};
+use chrono::{DateTime, Utc};
+use flate2::{read::ZlibDecoder, write::ZlibEncoder, Compression};
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
 use serde_with::*;
 use std::collections::HashMap;
+use std::io::{Read, Write};
 use unicode_normalization::UnicodeNormalization;
 
-use crate::data_macros::{impl_transparent_data_format_convertion, new_data_struct_type};
+use crate::data_macros::{impl_transparent_data_format_conversion, new_data_struct_type};
 use crate::ext_types::{new_uuid_type, DateTimeExtFormat};
 use crate::DeviceID;
-use parsec_api_crypto::{HashDigest, SecretKey};
+use parsec_api_crypto::{HashDigest, SecretKey, SigningKey, VerifyKey};
+
+pub trait CompSignEncrypt
+where
+    Self: Sized + Serialize + DeserializeOwned,
+{
+    fn author(&self) -> &DeviceID;
+    fn timestamp(&self) -> DateTime<Utc>;
+
+    fn check(
```

Comment:
Is there a reason to have this method out of `decrypt_verify_and_load` ?

Reply:
What's your suggestion ?